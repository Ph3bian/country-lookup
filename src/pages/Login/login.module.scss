@import "../../styles/generic";
.Login {
  display: block;


  @include md {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-row: 1;
    height: 100vh;
  }

  &Form {
    height: 100vh;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background: $primary;
    h2 {
      color: $secondary;
      text-align: center;
    }
    form {
      justify-content: center;
      button {
        width: 100%;
      }
    }
    &Container {
      max-width: 25rem;
      width: 20rem;
    }
  }
  &ImageHolder {
    display: none;
    @include md {
      display: block;
      &:before {
        content: "";
        display: block;
        height: 100%;
        top: 0;
        left: 0;
        width: 100%;
        background-color: rgba(0, 0, 0, 0.3);
      }
      width: 100%;
      background: center / cover no-repeat url("../../assets/images/world.jpg");
    }
  }
}
