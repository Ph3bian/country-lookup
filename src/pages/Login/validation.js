export default function validate(values) {
    let errors = {};
    if (!values.email) {
      errors.email = 'Email address is required';
    }   
     else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
      errors.email = 'Email address is invalid';
    }
    if (values.password.length < 6) {
        errors.password = "Minimum length must be greater than 6";
      }
      
    return errors;
  };