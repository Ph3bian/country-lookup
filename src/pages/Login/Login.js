import React, { useState } from 'react'
import PropTypes from 'prop-types';
import styles from './login.module.scss'
import { Input, Button } from 'components/Form'
import { useToasts } from 'components/Toaster'
import validation from './validation'
import Axios from 'utils/axios'

const Login = ({setAuth}) => {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [errors, setErrors] = useState({})
    const [loading, setLoading] = useState(false)
    const { addToast } = useToasts()

    const handleSubmit = e => {
        e.preventDefault()
        const errors = validation({ email, password })
        setErrors(errors)

        if (Object.keys(errors).length > 0) {
            return addToast('Oops! invalid credentials', {
                appearance: 'error'
            })
        }
        setLoading(true)

        return Axios.post(`/auth/login`, { email, password })
            .then(response => {
                addToast(response.data.message, { appearance: 'success' })
                const token = response.data.token
                setAuth(true)
                localStorage.setItem('countryToken', JSON.stringify(token))
                return setLoading(false)
            })
            .catch(({ response }) => {
                if (response) {
                    addToast(response.data[0].message, {
                        appearance: 'error'
                    })
                    return setLoading(false)
                } else {
                    addToast('Oops Something went wrong', {
                        appearance: 'error'
                    })
                    return setLoading(false)
                }
            })
    }

    return (
        <div className={styles.Login}>
            <div className={styles.LoginForm}>
                <form
                    className={styles.LoginFormContainer}
                    onSubmit={handleSubmit}
                >
                    <h2>Hello! login here</h2>
                    <Input
                        label={'Email'}
                        name={'email'}
                        type={'email'}
                        onChange={e => setEmail(e.target.value.toLowerCase())}
                        value={email}
                        error={errors.email}
                        required
                    />
                    <Input
                        label={'Password'}
                        name={'password'}
                        type={'password'}
                        onChange={e => setPassword(e.target.value)}
                        value={password}
                        error={errors.password}
                        required
                    />

                    <Button
                        type={'submit'}
                        value={'Submit'}
                        loading={loading}
                      
                    />
                </form>
            </div>
            <div className={styles.LoginImageHolder}></div>
        </div>
    )
}
Login.propTypes = {
    setAuth: PropTypes.func,
   
}
export default Login
