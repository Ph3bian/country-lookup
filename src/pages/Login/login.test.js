import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { ToastProvider } from 'react-toast-notifications'
import Login from './Login'

let container = null
beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
})
afterEach(() => {
    unmountComponentAtNode(container)
    container.remove()
    container = null
})

it('renders with or without setAuth', () => {
    act(() => {
        render(
            <ToastProvider>
                <Login />
            </ToastProvider>,
            container
        )
    })
    const header = container.querySelector('h2')
    expect(container.querySelector('form')).toBeDefined()
    expect(container.querySelector('button').innerHTML).toBe('Submit')
    expect(header.textContent).toBe('Hello! login here')
})

it('renders with props setAuth', () => {
    const props = { setAuth: jest.fn(() => false) }
    act(() => {
        render(
            <ToastProvider>
                <Login {...props} />
            </ToastProvider>,
            container
        )
    })
    

    expect(props.setAuth).toMatchSnapshot()
})
