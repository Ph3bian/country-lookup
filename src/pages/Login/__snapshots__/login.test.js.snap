// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props setAuth 1`] = `[MockFunction]`;
